### PR TITLE
Rename file share related classes

### PR DESCRIPTION
**Pull request by sima-zhu**
Finished the transformer work for FileRaneWriteType renaming. Thanks @alzimmermsft 
One more file name unchanged as transformer has some limitation on renaming. File an issue tracking this: https://github.com/Azure/azure-sdk-for-java/issues/6099
This issue got fixed with Garrett Serack's help.

### CODE REVIEW

File: sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareAsyncClient.java [hunk 1]
```diff
@@ -22,8 +22,8 @@
 import com.azure.storage.file.share.implementation.models.SharesCreateSnapshotResponse;
 import com.azure.storage.file.share.implementation.models.SharesGetPropertiesResponse;
 import com.azure.storage.file.share.implementation.models.SharesGetStatisticsResponse;
-import com.azure.storage.file.share.models.FileHttpHeaders;
-import com.azure.storage.file.share.models.FileSignedIdentifier;
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareAsyncClient.java]: `ShareFileSignedIdentifier`?

File: sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/implementation/FilesImpl.java [hunk 2]
```diff
@@ -108,7 +108,7 @@ public FilesImpl(AzureFileStorageImpl client) {
         @Put("{shareName}/{filePath}")
         @ExpectedResponses({201})
         @UnexpectedResponseExceptionType(FileStorageException.class)
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/implementation/FilesImpl.java]: Are we planning to change the name to `ShareFileStorageException` or `ShareStorageException`?

File: sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/implementation/models/ShareFileHttpHeaders.java [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+import com.azure.core.annotation.Fluent;
+import com.azure.core.implementation.util.ImplUtils;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlRootElement;
+
+/**
+ * Additional parameters for a set of operations, such as: Files_create,
+ * Files_setHTTPHeaders.
+ */
+@JacksonXmlRootElement(localName = "share-file-http-headers")
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/implementation/models/ShareFileHttpHeaders.java]: Should this be generating into `implementation`? I thought the previously named class was just in models, also does `FileHttpHeaders` need to be deleted and replaced with this?

File: sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/models/ShareFileHttpHeaders.java [hunk 4]
```diff
@@ -13,9 +13,9 @@
  * Additional parameters for a set of operations, such as: Files_create,
  * Files_setHTTPHeaders.
  */
-@JacksonXmlRootElement(localName = "file-http-headers")
+@JacksonXmlRootElement(localName = "share-file-http-headers")
 @Fluent
-public final class FileHttpHeaders {
+public final class ShareFileHttpHeaders {
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/models/ShareFileHttpHeaders.java]: I see this class showing up twice, once in `implementation.models` and here in `models`, was this one changed by hand or is the other a generation artifact from previous generations that had issues?

Comment (sima-zhu) [reply sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/models/ShareFileHttpHeaders.java]: Searched around the code. It is deleted

File: sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/implementation/FilesImpl.java (hunk 2 above)
Comment (sima-zhu) [reply sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/implementation/FilesImpl.java]: ShareStorageException now

File: sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareAsyncClient.java (hunk 1 above)
Comment (sima-zhu) [reply sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareAsyncClient.java]: Copied from .NET. This is used for all clients not only file client.

File: sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/implementation/models/ShareFileHttpHeaders.java (hunk 3 above)
Comment (sima-zhu) [reply sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/implementation/models/ShareFileHttpHeaders.java]: FileHttpHeaders.java has deleted. It is originally under model as it is a return type for public API. Forgot to delete the one under impl.

**Comment by sima-zhu** [comment]
The problem of FileDownloadHeaders got fixed. Closed the issue file above.

**Pull request merged**